lib/saml/settings_service.rb

```diff
@@ -1,28 +1,43 @@
 # frozen_string_literal: true
 require 'memoist'
+require 'sentry_logging'
 
 module SAML
   # This class is responsible for putting together a complete ruby-saml
   # SETTINGS object, meaning, our static SP settings + the IDP settings
   # which must be fetched once and only once via IDP metadata.
   class SettingsService
     class << self
+      include SentryLogging
       extend Memoist
 
       METADATA_RETRIES = 3
       OPEN_TIMEOUT = 2
       TIMEOUT = 15
 
       def saml_settings
+        if @fetch_attempted.nil? || metadata_successfully_retrieved?
+          merged_saml_settings
+        else
+          # retrieve was attempted but failed, blow away memoized data and try again
+          merged_saml_settings(true)
```

Comment:
Couldn't this also be done with a Singleton? Upon initialization the singleton would attempt to fetch metadata using a worker (retry enabled), if successful it queues up a fresh pull for the following day so it is always constantly updating the RubySaml::Settings with retry builtin. By nature of being a singleton it automatically always going to be in memory.